packages/CatalogService/src/apiFactory/operations/catalogFactory.test.ts

```diff
@@ -0,0 +1,158 @@
+import {
+  QueryResponse,
+  RDFSResource,
+  RDFSResourceDescendant,
+} from "@telicent-oss/rdfservice";
+import { Console } from "console";
+import {
+  CatalogService,
+  DCATCatalog,
+  DCATDataService,
+  DCATDataset,
+  DCATResource,
+  DcatResourceQuerySolution,
+} from "../../../index";
+
+import { catalogFactory } from "./catalogFactory";
+import {
+  ADDITIONAL_DATA_SET,
+  ADDITIONAL_RESULT,
+  BASIC_DATA_SET,
+  BASIC_RESULT,
+  CATALOG_QUERY_TEMPLATE,
+  OWNER2_CATALOG_ITEM,
+  OWNER2_RESULT,
+  OWNER_CATALOG_ITEM,
+  OWNER_OF_ALL_CATALOG_ITEM,
+  OWNER_OF_ALL_RESULT,
+  OWNER_RESULT,
+  QUERY_STRING,
+  TEMPLATE_RESULT,
+} from "./mock_catalog_query_data";
+class MockCatalogService {
+  runQuery(query: string) {
+    return {};
+  }
+}
+const runQuerySpy = jest.spyOn(MockCatalogService.prototype, "runQuery");
+
+beforeEach(() => {
+  runQuerySpy.mockClear();
+});
+
+const setupServiceData = async (
+  data: QueryResponse
+): Promise<MockCatalogService> => {
+  runQuerySpy.mockImplementation(async (_) => {
+    return await data;
+  });
+  const service = new MockCatalogService();
+  return service;
+};
+
+test("Check empty result for catalog", async () => {
+  const mockService = await setupServiceData(CATALOG_QUERY_TEMPLATE([]));
+  const sf = catalogFactory(mockService as unknown as CatalogService);
+  const EXPECTED_RESULT = [TEMPLATE_RESULT()];
+  const results = await sf({ dataResourceFilters: ["all"], searchText: "" });
+  expect(results).toStrictEqual(EXPECTED_RESULT);
```

Comment:
Suggestion: Put the results inline
Rationale: generating expected outputs can leave the door open for test bugs , and being explicit with tests inline makes them easier to read (esp in github web ui)
```suggestion
  expect(results).toMatchInlineSnapshot();
```

Reply:
or both